lib/explorer/series.ex

```diff
@@ -1274,13 +1274,19 @@ defmodule Explorer.Series do
 
   """
   @doc type: :element_wise
-  def categorise(%Series{dtype: l_dtype} = series, %Series{dtype: dtype} = categories)
-      when K.and(K.in(l_dtype, [:string | @integer_types]), K.in(dtype, [:string, :category])),
+  def categorise(%Series{dtype: l_dtype} = series, %Series{dtype: :category} = categories)
+      when K.in(l_dtype, [:string | @integer_types]),
       do: apply_series(series, :categorise, [categories])
 
+  def categorise(%Series{dtype: l_dtype} = series, %Series{dtype: :string} = categories)
+      when K.in(l_dtype, [:string | @integer_types]) do
+    categories = categories |> distinct() |> cast(:category)
+    apply_series(series, :categorise, [categories])
+  end
+
```

Comment:
Moved string series and list of strings categorise to here. 
 - instead of throwing errors, it applies distinct to categories series
 - nil check is missing - should we add nil_count  check ?

Reply:
> instead of throwing errors, it applies distinct to categories series

I don't think we should do this because we are mapping indexes into the list. If you remove duplicates, the indexes are shifted, and the result changes. Is there a reason we removed the Rust code responsible for this?

Reply:
Hit a wall with getting RevMapping to work in Series.Categorise - so tried if it can be fixed in elixir

Reply:
we can do it in Elixir, but we need to get the unique_count and raise if different than size, and the nil count and raise.